server/src/main/java/org/jboss/as/server/ApplicationServerService.java

```diff
@@ -109,6 +111,20 @@ public synchronized void start(final StartContext context) throws StartException
             b.append(ServerLogger.ROOT_LOGGER.configuredSystemPropertiesLabel());
             for (String property : new TreeSet<String>(properties.stringPropertyNames())) {
                 String propVal = property.toLowerCase(Locale.getDefault()).contains("password") ? "<redacted>" : properties.getProperty(property, "<undefined>");
+                if (property.toLowerCase(Locale.getDefault()).equals("sun.java.command") && !propVal.isEmpty()) {
```

Comment:
I wonder if 1 we should care about the property and 2 if we should have a more generic way of checking for other keywords where the password might not want to be printed.

Reply:
@jamezp 
>I wonder if 1 we should care about the property
"sun.java.command" exposes all system properties and makes password masking on individual properties obsolete.    
>2 if we should have a more generic way of checking for other keywords where the password might not want to be printed.
How to determine that a property is a password type?

Reply:
@ivassile It seems like bad practice to me to pass a password via the command line. That said my thought was to maybe have some kind of map or set of possible system properties which could be filtered. I feel like we've done this elsewhere, but I haven't had a chance to look.

This is just more of an idea though and I could be way off basis.

Reply:
That fix fulfilled the request from the Red Hat support folks who wanted it. I deliberately kept it simple as the whole problem can be avoided by turning the level of the log category to INFO.

I see WFCORE-4680 as a bug fix correcting a hole in my original work. Configuration options to control this would be a separate RFE, IMO.

Reply:
This makes sense to me then. I just didn't know if we had random other properties we wanted replaced. I hadn't realized this was a follow up of the original.